### DISCUSSION TITLE
`.codespell.exclude` outdated?

### DISCUSSION THREAD

**Discussion by TheGiraffe3** [endless-sky/endless-sky#10797] [General]
For ignoring specific words on specific lines, we currently have a `.codespell.exclude` file in which we put the exact line that must not be checked.
Recently, when I was making a pull request to add a spellcheck to Naev, bobbens pointed out that a `.codespell.exclude` file is unnecessary, and will cause unneeded conflicts.
Apparently, within the last year, the codespell-project people merged a PR that allows for ignoring specific _words_ in specific lines, _in the file itself_.

I was wondering if Endless Sky wanted to also begin using that approach, or if deleting `.codespell.exclude` would cause too many merge conflicts to be worth it.

Here's the PR link: https://github.com/codespell-project/codespell/pull/2400

**Comment by tibetiroka** [comment]
We have a codespell.words.exclude file for collecting common words. I'm not a huge fan of the inline includes tbh, as their location aren't apparent once the content has changed enough.

**Reply by TheGiraffe3** [reply]
What do you mean by "location aren't apparent"? It's still on the line with the "misspelled" word.

The main advantages to switching would be no more merge conflicts every time something modifying the file is squashed in (and also it's easier to see where each "misspelled" word is from).
But switching may cause conflicts for a bit.

**Reply by tibetiroka** [reply]
> What do you mean by "location aren't apparent"? It's still on the line with the "misspelled" word.

Do we have to write it on every line where that word is used? That's going to be rather cumbersome.

If we only have to write it once per file, it's not going to be apparent for reviewers whether a file already has that word excluded somewhere earlier.

We also run into cases where new words get added to codespell and we have to adjust our previous exclusions; this would become more difficult if we have to do it on every line, instead of applying it globally.

**Reply by TheGiraffe3** [reply]
We keep the `.codespell.words.exclude` file.
But for the lines in `.codespell.exclude`, we instead specify to ignore the "misspelled" word on the line.
Because the .codespell.exclude isn't global, just for the one line, it wouldn't be too difficult to do.

**Reply by TheGiraffe3** [reply]
Well, here's an example branch doing this: https://github.com/TheGiraffe3/endless-sky/tree/remove-codespell.exclude
But now that I look at the error list... well, it's something to consider, but you're right, it's probably more difficult to change to after having the other established for so long.

**Comment by ziproot** [comment]
I think some of the words (aesthetic, aesthetics, aesthetically) can be removed if Endless Sky uses the latest version of codespell, as that no longer flags those words.
